Assets/Mirage/Runtime/Transport/SocketLayer/ISocket.cs

```diff
@@ -0,0 +1,50 @@
+using System.Net;
+
+namespace Mirage.SocketLayer
+{
+    /// <summary>
+    /// Link between Mirage and the outside world
+    /// </summary>
+    public interface ISocket
+    {
+        /// <summary>
+        /// Starts listens for data on an endpoint
+        /// <para>Used by Server to allow clients to connect</para>
+        /// </summary>
+        /// <param name="endPoint">the endpoint to listen on</param>
+        void Bind(EndPoint endPoint);
+
+        /// <summary>
+        /// Closes the socket, stops receiving messages from other peers
+        /// </summary>
+        void Close();
+
+        /// <summary>
+        /// Checks if a packet is available 
+        /// </summary>
+        /// <returns>true if there is atleast 1 packet to read</returns>
+        bool Poll();
+
+        /// <summary>
+        /// Gets next packet
+        /// <para>Should be called after Poll</para>
+        /// <para>
+        ///     Implementation should check that incoming packet is within the size of <paramref name="buffer"/>,
+        ///     and make sure not to return <paramref name="bytesReceived"/> above that size
+        /// </para>
+        /// </summary>
+        /// <param name="buffer">buffer to write recevived packet into</param>
+        /// <param name="endPoint">where packet came from</param>
+        /// <param name="bytesReceived">length of packet, should not be above <paramref name="buffer"/> length</param>
+        void Receive(byte[] buffer, ref EndPoint endPoint, out int bytesReceived);
```

Comment:
Would be better to have it as `int Receive` and return the amount of received bytes, for consistency with .NET socket.